Mathlib/Algebra/BigOperators/Finsupp/Basic.lean

```diff
@@ -405,6 +405,12 @@ theorem univ_sum_single_apply' [AddCommMonoid M] [Fintype α] (i : α) (m : M) :
   classical rw [Finset.sum_pi_single]
   simp
 
+lemma sum_single_add_single (f₁ f₂ : ι) (g₁ g₂ : A) (F : ι → A → B) (H : f₁ ≠ f₂)
+    (HF : ∀ f, F f 0 = 0) :
+    sum (single f₁ g₁ + single f₂ g₂) F = F f₁ g₁ + F f₂ g₂ := by
+  classical
+  simp +contextual [sum_of_support_subset _ (s := {f₁, f₂}) (support_add.trans _),
+    Finset.subset_iff, single_apply, or_imp, H, HF, H.symm]
```

Comment:
```suggestion
  simp +contextual [sum_of_support_subset _ support_single_add_single_subset,
    Finset.subset_iff, single_apply, or_imp, H, HF, H.symm]
```
You can probably do this and remove some of the lemmas on the next line. Potentially even `+contextual`.